addons/payment_authorize/controllers/main.py

```diff
@@ -36,6 +36,8 @@ def authorize_form_feedback(self, **post):
     def authorize_s2s_create_json(self, **kwargs):
         acquirer_id = int(kwargs.get('acquirer_id'))
         acquirer = request.env['payment.acquirer'].browse(acquirer_id)
+        if not kwargs.get('partner_id'):
+            kwargs = dict(kwargs, partner_id=request.env['res.users'].browse(request.env.uid).partner_id.id)
```

Comment:
use  partner_id=request.env.user.partner_id.id

Reply:
Indeed you directly have the partner form the user record in the environment. Just triple-check there is no access right issues.